### PR TITLE
Make process_changes return a set of what actually changed

### PR DESCRIPTION
**Pull request by bdraco**
I realized we still write state too frequently. When we do the poll if the value is the same we still callback the entity.

Will be used in https://github.com/home-assistant/core/pull/102370

### CODE REVIEW

File: aiohomekit/model/__init__.py [hunk 1]
```diff
@@ -407,16 +407,29 @@ def has_aid(self, aid: int) -> bool:
...
+    ) -> set[tuple[int, int]]:
+        """Process changes from a HomeKit controller.
+
+        Returns a set of the changes that were applied.
+        """
+        changed: set[tuple[int, int]] = set()
+        for aid_iid, value in changes.items():
+            (aid, iid) = aid_iid
             if not (char := self.aid(aid).characteristics.iid(iid)):
                 continue
```
Comment (Jc2k) [inline aiohomekit/model/__init__.py:422]: Should we just always add it to changed if it's one of the event characteristics?

Comment (bdraco) [reply aiohomekit/model/__init__.py:422]: Yes we will need to do that

Comment (bdraco) [reply aiohomekit/model/__init__.py:423]: https://github.com/Jc2k/aiohomekit/pull/339/files#diff-5443a6f050cba5fd14b7604e81e8139f66401a7f2eb74c8e8f396837127b2b3cR425

**Review by Jc2k — APPROVED** [review]

**Comment by bdraco** [comment]
thanks, will release shortly

**Pull request merged**